libfreerdp/core/certificate.c

```diff
@@ -425,21 +426,29 @@ static BOOL certificate_process_server_public_signature(rdpCertificate* certific
 	if (sum != 0)
 	{
 		WLog_ERR(TAG, "invalid signature");
-		//return FALSE;
+		return FALSE;
 	}
 
 	siglen -= 8;
 
-	// TODO: check the result of decrypt
-	crypto_rsa_public_decrypt(encsig, siglen, TSSK_KEY_LENGTH, tssk_modulus, tssk_exponent, sig);
+	if (crypto_rsa_public_decrypt(encsig, siglen, TSSK_KEY_LENGTH, tssk_modulus, tssk_exponent,
```

Comment:
I remember of a check that was not done on purpose as sometime the server sends garbage. I'm wondering if it's not that one, @awakecoding any opinion on this ?